packages/alias/src/index.js

```diff
@@ -99,39 +87,13 @@ export default function alias(options = {}) {
         return customResolver(updatedId, importerId);
       }
 
-      if (isFilePath(updatedId)) {
-        const directory = posix.dirname(importerId);
-
-        // Resolve file names
-        const filePath = posix.resolve(directory, updatedId);
-        const match = resolve
-          .map((ext) => (endsWith(ext, filePath) ? filePath : `${filePath}${ext}`))
-          .find(exists);
-
-        if (match) {
-          updatedId = match;
-          // To keep the previous behaviour we simply return the file path
-          // with extension
-        } else if (endsWith('.js', filePath)) {
-          updatedId = filePath;
-        } else {
-          const indexFilePath = posix.resolve(directory, `${updatedId}/index`);
-          const defaultMatch = resolve.map((ext) => `${indexFilePath}${ext}`).find(exists);
-          if (defaultMatch) {
-            updatedId = defaultMatch;
-          } else {
-            updatedId = `${filePath}.js`;
-          }
+      return this.resolve(updatedId, importer, { skipSelf: true }).then(resolved => {
```

Comment:
does `this.resolve` go through other plugins or does it bypass them?

Reply:
It goes through other plugins. Through all plugins except current - alias plugin (`skipSelf:true`). This allow other plugins to resolve `updatedId` if they can.